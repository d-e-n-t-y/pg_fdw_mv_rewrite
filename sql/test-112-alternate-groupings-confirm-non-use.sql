-- Anticipate no rewrite
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN explain (VERBOSE, COSTS OFF)
	select 'key=' || key, COUNT (value) from test group by 'key=' || key
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;

select 'key=' || key, COUNT (value) from test group by 'key=' || key;

-- Anticipate no rewrite
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN explain (VERBOSE, COSTS OFF)
	select 'key=' || substr (key, 1, 3), COUNT (value) from test group by 'key=' || substr (key, 1, 3)
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;

select 'key=' || substr (key, 1, 3), COUNT (value) from test group by 'key=' || substr (key, 1, 3);
