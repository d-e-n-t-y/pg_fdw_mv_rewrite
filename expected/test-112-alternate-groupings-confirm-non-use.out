-- Anticipate no rewrite
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN explain (VERBOSE, COSTS OFF)
	select 'key=' || key, COUNT (value) from test group by 'key=' || key
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;
select 'key=' || key, COUNT (value) from test group by 'key=' || key;
 ?column? |  count  
----------+---------
 key=key1 |      10
 key=key2 | 1000000
(2 rows)

-- Anticipate no rewrite
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN explain (VERBOSE, COSTS OFF)
	select 'key=' || substr (key, 1, 3), COUNT (value) from test group by 'key=' || substr (key, 1, 3)
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;
select 'key=' || substr (key, 1, 3), COUNT (value) from test group by 'key=' || substr (key, 1, 3);
 ?column? |  count  
----------+---------
 key=key  | 1000010
(1 row)

